.github/workflows/coverage.yml

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) Prevail Verifier contributors.
+# SPDX-License-Identifier: MIT
+
+name: CPP Code Coverage
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build_ubuntu:
+    strategy:
+      matrix:
+        configurations: [Debug, Release]
+    runs-on: ubuntu-20.04
+    env:
+      # Configuration type to build.  For documentation on how build matrices work, see
+      # https://docs.github.com/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
+      BUILD_CONFIGURATION: ${{matrix.configurations}}
+
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo apt install libboost-dev libyaml-cpp-dev
+          pip install gcovr
+
+      - uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+
+      - name: Build
+        run: |
+          mkdir build
+          cmake -B build -DCMAKE_BUILD_TYPE=${{env.BUILD_CONFIGURATION}} -DCMAKE_CXX_FLAGS="--coverage -g -O0"
+          cmake --build build -j $(nproc)
+
+      - name: Run unit tests
+        run: ./tests -d yes
+
+      - name: Generate code coverage report
+        run: |
+          cd build
+          gcovr -r .. --cobertura ../ebpf-verifier.xml --gcov-ignore-parse-errors
+
+      - name: Upload Report to Codecov
+        uses: codecov/codecov-action@f32b3a3741e1053eb607407145bc9619351dc93b
```

Comment:
What is this token?

Reply:
That is the sha256 hash of the version of the CodeCov action. By pinning to a specific version of the task we avoid supply chain attacks.